roles/golang/README.md

```diff
@@ -0,0 +1,85 @@
+# golang
+
+This role installs a specific version of the Go toolchain from the official
+`go.dev` tarballs into `/usr/local/go`, and ensures `go` is available on
+`$PATH` via `/usr/local/bin/go`.
+
+---
+
+## What it does
+
+On each run, the role:
+
+1. Ensures the download directory (`{{ golang_download_dir }}`) exists.
+2. Derives the correct architecture string (`amd64` / `arm64`) from
+   `ansible_architecture` if you didn’t override `golang_arch`.
```

Comment:
Why would I want to override `golang_arch`? Shouldn't the discovery of the architecture always work?